arrows/core/transform_detected_object_set.cxx

```diff
@@ -0,0 +1,279 @@
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "transform_detected_object_set.h"
+
+#include <iostream>
+#include <vital/io/camera_io.h>
+#include <vital/config/config_difference.h>
+#include <vital/util/string.h>
+#include <vital/types/bounding_box.h>
+#include <Eigen/Core>
+
+namespace kwiver {
+namespace arrows {
+namespace core {
+
+// ------------------------------------------------------------------
+transform_detected_object_set::transform_detected_object_set()
+  : src_camera_krtd_file_name( "" )
+  , dest_camera_krtd_file_name( "" )
+{
+}
+
+// ------------------------------------------------------------------
+transform_detected_object_set::transform_detected_object_set(kwiver::vital::camera_perspective_sptr src_cam,
+							     kwiver::vital::camera_perspective_sptr dest_cam)
+  : src_camera( src_cam )
+  , dest_camera( dest_cam )
+{
+}
+
+// ------------------------------------------------------------------
+vital::config_block_sptr
+transform_detected_object_set::get_configuration() const
+{
+  // Get base config from base class
+  vital::config_block_sptr config = vital::algorithm::get_configuration();
+
+  config->set_value( "src_camera_krtd_file_name", src_camera_krtd_file_name,
+                     "Source camera KRTD file name path" );
+
+  config->set_value( "dest_camera_krtd_file_name", dest_camera_krtd_file_name,
+                     "Destination camera KRTD file name path" );
+
+  return config;
+}
+
+
+// ------------------------------------------------------------------
+void
+transform_detected_object_set::
+set_configuration( vital::config_block_sptr config_in )
+{
+  vital::config_block_sptr config = this->get_configuration();
+
+  config->merge_config( config_in );
+  this->src_camera_krtd_file_name = config->get_value< std::string > ( "src_camera_krtd_file_name" );
+  this->dest_camera_krtd_file_name = config->get_value< std::string > ( "dest_camera_krtd_file_name" );
+
+  this->src_camera = kwiver::vital::read_krtd_file( this->src_camera_krtd_file_name );
+  this->dest_camera = kwiver::vital::read_krtd_file( this->dest_camera_krtd_file_name );
+}
+
+
+// ------------------------------------------------------------------
+bool
+transform_detected_object_set::
+check_configuration( vital::config_block_sptr config ) const
+{
+  kwiver::vital::config_difference cd( this->get_configuration(), config );
+  const auto key_list = cd.extra_keys();
+
+  if ( ! key_list.empty() )
+  {
+    LOG_WARN( logger(), "Additional parameters found in config block that are not required or desired: "
+              << kwiver::vital::join( key_list, ", " ) );
+  }
+
+  return true;
+}
+
+
+// ------------------------------------------------------------------
+Eigen::Vector3d
+transform_detected_object_set::
+backproject_to_height( const kwiver::vital::camera_perspective_sptr camera,
```

Comment:
This function is actually a special case of `backproject_to_plane` where plane = [0, 0, 1, 0].  So you can probably get rid of this one and just use `backproject_to_plane`, or have this function call `backproject_to_plane` to do the work.  If you are keeping this function and not making `height` a parameter I would call it `backproject_to_ground`.